test/functional/feature_proxy.py

```diff
@@ -263,12 +267,13 @@ def networks_dict(d):
 
         n2 = networks_dict(self.nodes[2].getnetworkinfo())
         assert_equal(NETWORKS, n2.keys())
+        proxy = f'{self.conf2.addr[0]}:{self.conf2.addr[1]}'
         for net in NETWORKS:
             if net == NET_I2P:
                 expected_proxy = ''
                 expected_randomize = False
             else:
-                expected_proxy = f'{self.conf2.addr[0]}:{self.conf2.addr[1]}'
+                expected_proxy = proxy
```

Comment:
nit: this change looks unnecessary because the expression is used just once, so putting it in a variable does not make it more maintainable. It maybe makes it less readable because now one has to visually go up a few lines in order to find what `expected_proxy` ends up with.

Reply:
This line is invoked 4 times in the loop, so it seems better to do the string interpolation before entering it.

Reply:
Oh, performance :eyes: :space_invader: